web/admin/src/RightReportsTable.js

```diff
@@ -22,23 +22,27 @@ export default class RightReportsTable extends Component {
   constructor(props) {
     super(props)
     this.state = {
+      isShowingApproved: false
     }
   }
 
 
   render() {
     const { totalBlocked, itemsAndReports, getUser, getReport, returnItem, returnContent, markBlock, approveQ, blockAll, approveAll, unBlock } = this.props
-
     return (
       <div className="questionBox2">
-      <div className="cellBoxTop">
-        <p className="listTitle">Blocked ({totalBlocked})</p>
-      </div>
+      {this.renderHeaderBox()}
       <ul className='listBox2' ref={(input) => {this.flaggedList = input}}>
         { itemsAndReports.map((itemAndReport) => {
           const id = itemAndReport.item.id
-          const allReportsFlagged = Object.values(itemAndReport.reports).filter(item => item.isBlock === true && item.approved !== true)
-          if (allReportsFlagged.length) {
+          let allReports = []
```

Comment:
Don't initialize `allReports` to `[]` since it always gets a value that overwrites it.  Simply `let allReports`

Reply:
Alternatively
```javascript
const allReportsFilter = this.state.isShowingApproved
  ? item => item.isBlock === false && item.isApproved === true
  : item => item.isBlock === true && item.isApproved !== true
const allReports = Object.values(itemAndReport.reports).filter(allReportsFilter)
```